gobblin-core/src/main/java/gobblin/writer/http/AsyncHttpWriterBuilder.java

```diff
@@ -72,7 +83,23 @@
   }
 
   AsyncHttpWriterBuilder<D, RQ, RP> fromState(State state) {
-    this.state = state;
+    if (!(state instanceof WorkUnitState)) {
+      throw new RuntimeException(String.format("AsyncHttpWriterBuilder requires a %s on construction.", WorkUnitState.class.getSimpleName()));
+    }
+
+    this.state = (WorkUnitState) state;
+    try {
+      if (state.getPropAsBoolean (ConfigurationKeys.WRITER_LIMIT_ENABLED_KEY, ConfigurationKeys.DEFAULT_WRITER_LIMIT_ENABLED)) {
```

Comment:
The point of the broker is that you don't need to handle configuration yourself, you simply use the `SharedLimiterFactory`, and the factory will automatically create a Noop if necessary (e.g. if no limiter is defined). Because of this, there's no need to have a separate configuration key and check to see if the limiter is enabled.